src/js/angular/clustermanagement/services/cluster-drawing.service.js

```diff
@@ -294,6 +294,23 @@ function calculateElementSizeByText(text) {
     return {height, width};
 }
 
+function addRecoveryLabelListeners(object, message, width, height, displayMessage) {
+    object.on('mouseover.tooltip', () => {
+        object
+            .attr('width', calculateElementSizeByText(message).width)
+            .attr('height', calculateElementSizeByText(message).height)
+            .attr('x', -calculateElementSizeByText(message).width / 2)
```

Comment:
Is there some way to skip calculating the same thing twice?

Reply:
Possibly. Assigning them to a field would need to happen within the listener, or maybe in the method which calls this one.